sale_automatic_workflow/sale.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+#    sale_automatic_workflow for OpenERP
+#    Copyright (C) 2011 Akretion Sébastien BEAU <sebastien.beau@akretion.com>
+#    Copyright 2013 Camptocamp SA (Guewen Baconnier)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+from openerp import models, fields, api, _
+
+
+class sale_order(models.Model):
+    _inherit = "sale.order"
+
+    workflow_process_id = fields.Many2one(comodel_name='sale.workflow.process',
+                                          string='Automatic Workflow',
+                                          ondelete='restrict')
+
+    def _prepare_invoice(self, cr, uid, order, lines, context=None):
+        invoice_vals = super(sale_order, self)._prepare_invoice(
+            cr, uid, order, lines, context=context)
+        workflow = order.workflow_process_id
+        if not workflow:
+            return invoice_vals
+        invoice_vals['workflow_process_id'] = workflow.id
+        if workflow.invoice_date_is_order_date:
+            invoice_vals['date_invoice'] = order.date_order
+        if workflow.property_journal_id:
+            invoice_vals['journal_id'] = workflow.property_journal_id.id
+        return invoice_vals
+
+    def _prepare_order_picking(self, cr, uid, order, context=None):
```

Comment:
`_prepare_order_picking` does no longer exist in 9.0.
You need something like that:

```python
class StockMove(models.Model):
    _inherit = 'stock.move'

    @api.model
    def _prepare_picking_assign(self, move):
        """ Prepares a new picking for this move as it could not be assigned to
        another picking. This method is designed to be inherited.
        """
        values = super(StockMove, self)._prepare_picking_assign(move)
        if move.procurement_id.sale_line_id:
            sale = move.procurement_id.sale_line_id.order_id
            values['workflow_process_id'] = sale.workflow_process_id.id
        return values

```